### PR TITLE
✨ : cost of ownership

### LINKED ISSUE #43: ✨ : add estimated cost-ownership on modules
for cloud-type modules, an estimated cost of ownership could be helpful for the users.

### PR DESCRIPTION
**Pull request by juwit**
resolves #43

### CODE REVIEW

**Comment by cdubuisson** [comment]
there is a typo in the last commit message: `unsed` instead of `unused`

**Comment by cdubuisson** [comment]
I have met a exception in the stack screen if field estimatedMonthlyCost is not filled

File: src/main/java/io/codeka/gaia/service/StackCostCalculator.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                start = LocalDateTime.now();
+                end = LocalDateTime.now();
+            }
+        }
+
+        // add last duration
+        duration = duration.plus(Duration.between(start, end));
+
+        // get hourly cost
+        var hourlyCost = module.getEstimatedMonthlyCost().divide(BigDecimal.valueOf(31L*24L), 4, RoundingMode.HALF_UP);
```
Comment (cdubuisson) [inline src/main/java/io/codeka/gaia/service/StackCostCalculator.java]: if the field is not filled, an exception is thrown when accessing the stack screen

File: src/main/java/io/codeka/gaia/controller/StackController.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ public String editStack(@PathVariable String stackId, Model model){
         return "stack";
     }
 
-    @GetMapping("/stacks/{stackId}/apply")
+    @GetMapping("/stacStackks/{stackId}/apply")
```
[the remark this answers is not in the document]
Comment (juwit) [reply src/main/java/io/codeka/gaia/controller/StackController.java]: Yes that what it is !
I guess that also means that there are not enough tests on this part !

File: src/main/java/io/codeka/gaia/service/StackCostCalculator.java (hunk 1 above)
Comment (juwit) [reply src/main/java/io/codeka/gaia/service/StackCostCalculator.java]: > I notice a difference in the render of the module description page as well.
> The padding which was previously on the readme part is no more.
> And the tabs look bigger too.
> 
> Screenshot_2019-08-09 Gaia - before
> 
> Screenshot_2019-08-09 Gaia - after

I've corrected the size of the text, and added padding to the readme tab using the same <b-container fluid> as I did in the cost-ownership tab.

File: src/main/java/io/codeka/gaia/controller/StackController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (juwit) [reply src/main/java/io/codeka/gaia/controller/StackController.java]: Correction done.

File: src/main/java/io/codeka/gaia/service/StackCostCalculator.java (hunk 1 above)
Comment (juwit) [reply src/main/java/io/codeka/gaia/service/StackCostCalculator.java]: Correction done.

File: src/main/resources/templates/vue_templates/readme.vue [hunk 3]
```diff
@@ -1,10 +1,10 @@
 <template id="readme">
-    <div>
+    <b-container fluid>
         <b-spinner label="Loading..." class="m-5" style="height: 2.5rem; width: 2.5rem" v-if="!loaded"></b-spinner>
         <div class="markdown-body" v-if="loaded">
             <div v-html="content"></div>
         </div>
-    </div>
+    </b-container>>
```
Comment (cdubuisson) [inline src/main/resources/templates/vue_templates/readme.vue]: small typo here which generates a big stacktrace in console 
```suggestion
    </b-container>
```

**Pull request merged**